tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,247 @@
+            }
+        }
+    }
+}
+//////////////////
+// THROW ERRORS //
+//////////////////
+if (!empty($errors)) {
+    print_r($errors);
+    die("Resolve errors and run script again.\n");
+} else {
+    echo "\nValidation successful.\n";
+}
+
+/////////////////////
+// CONTINUE SCRIPT //
+/////////////////////
+
+// Update ConfigSetting table with value of 'useConsent' if true. Default is set to false.
+$configID = $db->pselectOne(
+    'SELECT ID FROM ConfigSettings WHERE Name="useConsent"',
+    array()
+);
+if ($useConsent === "true") {
+    $updateValue = ['Value' => $useConsent];
+    $db->update(
+        'Config',
+        $updateValue,
+        array (
+         'ConfigID' => $configID, 
+        )
+    );
+    echo "\nConfig settings set to use consent.\n\n";
+}
+
+/////////////////////////////////////////////
+// START IMPORT OF CONSENT INTO NEW TABLES //
+/////////////////////////////////////////////
+
+$consentType = [];
+$printArray  = [];
+foreach ($consents as $key=>$consent) {
+    $consentName       = $consent['name'];
+    $consentLabel      = $consent['label'];
+  
+    // Populate consent table with consents from Config.xml
+    $db->insert(
+        'consent', 
+        array(
+           'Name'  => $consentName,
+           'Label' => $consentLabel,
+         )
+    );
+    // Save ConsentID inserted
+    $consentID = $db->pselectOne(
+        'SELECT ConsentID FROM consent WHERE Name=:consentName',
+        array('consentName' => $consentName)
+    );
+    // Create array of consents to use later in importing history data
+    $consentList[$consentName] = $consentLabel;
+   
+    // Get all data where consent status has a value in participant_status
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $dataQuery         = 'SELECT CandID, ' . $consentName . ', ' . $consentDate . ', ' . $consentWithdrawal . ' FROM participant_status 
+                         WHERE ' . $consentName . ' IS NOT NULL OR ' . $consentName . ' != ""';
+    $psData            = $db->pselect($dataQuery, array());
+    print_r($psData);
+    foreach ($psData as $entry) {
+        // Import 'not_answered' status as NULL
+        if ($entry[$consentName] === "not_answered") {
+            $entry[$consentName] = "";
+        }
+        // Format data
+        $consentValues = [
+            'CandidateID'   => $entry['CandID'],
+            'ConsentID'     => $consentID,
+            'Status'        => $entry[$consentName],
+            'DateGiven'     => $entry[$consentDate],
+            'DateWithdrawn' => $entry[$consentWithdrawal],
+        ];
+        // Push each formatted entry to array
+        array_push($printArray, $consentValues);
+    }
+}
+
+// Output list of data to be inserted into new table
+echo "\nRows to be inserted into 'candidate_consent_rel' table ..\n";
+echo "Empty values will be inserted as NULL.\n\n";
```

Comment:
does this mean the empty array, or does it mean if at least one column is empty? (If the former, shouldn't it just be skipped?)